### DISCUSSION TITLE
Issues with latest CefSharp and Windows 11 22H2

### DISCUSSION THREAD

**Discussion by jhtony** [cefsharp/CefSharp#4956] [Q&A]
I am looking for guidance as to what would cause CefSharp.Common and CefSharp.Wpf to run into crashing issues when running on a machine that is running Windows 11 22H2 and only that specific version.  We are not seeing any issues with this running on any version of Windows 10 or other versions of Windows 11.  We are currently running version 129.0.110 for both of these NuGet packages.  We have tried to remedy this using older versions of the CefSharp.Common/CefSharp.Wpf but there was no difference in behavior with those older releases.  

Several customers and internal testers were seeing our 32-bit application crash when opening multiple CefSharp-based browser tabs. Several of our developers could not duplicate the issue on devices that were running Windows 11 Version 21H2. Further investigation found that the people who were seeing the issue were on Windows 11 Version 22H2, or to a lesser extent Version 23H2.

A developer updated his device to Windows 11 Version 22H2 and was then able to duplicate the issue whenever five or six browser tabs were opened in the application. The following is an example of the entry we’d see in the EventViewer:

System.Runtime.InteropServices.COMException (0x88980003): MILERR_WIN32ERROR (Exception from HRESULT: 0x88980003)
   at System.Windows.Media.Imaging.WriteableBitmap..ctor(Int32 pixelWidth, Int32 pixelHeight, Double dpiX, Double dpiY, PixelFormat pixelFormat, BitmapPalette palette)
   at CefSharp.Wpf.Rendering.WritableBitmapRenderHandler.<>c__DisplayClass8_0.<CreateOrUpdateBitmap>b__0() in C:\projects\cefsharp\CefSharp.Wpf\Rendering\WritableBitmapRenderHandler.cs:line 145
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

Within the CefSharp browser code we added additional garbage collection where appropriate in addition to changing the CefSharp settings to minimize GPU resource usage. None of these changes affected the crashing issue.

We also made changes to the registry:

1. Added IRPStackSize and set to 100.
2. Bumped GDIProcessHandleQuota to 65.535; UserProcessHandleQuota to 18,00 in two places for each (Windows and Windows NT)

These registry changes also did not change the crashing issue.

Thanks in advance!

**Comment by amaitland** [comment]
> Several customers and internal testers were seeing our 32-bit application crash when opening multiple CefSharp-based browser tabs

Using the 32bit version of the wpf implementation is really not recommended, you'll very easily run out of memory with lots of tabs, specially if you are using a larger display e.g. 4k

Make sure your executable is large address aware.
https://github.com/cefsharp/CefSharp/wiki/General-Usage#win32-out-of-memory

Alternative is to use https://github.com/cefsharp/CefSharp.Wpf.HwndHost which embeds the native browser

**Reply by amaitland** [reply]
If you have to use the 32bit version then you may need to write your own rendering implement and more aggressively free the memory.

https://github.com/dotnet/wpf/issues/3067

**Comment by jhtony** [comment]
Thank you for the response.  We will give these options a try and see if it remedies the issue on the Windows 11 22H2 machines.
